lightning/src/ln/channel.rs

```diff
@@ -2969,6 +2969,30 @@ impl<Signer: Sign> Channel<Signer> {
 			panic!("Cannot update fee while peer is disconnected/we're awaiting a monitor update (ChannelManager should have caught this)");
 		}
 
+		// Before to propose a feerate update, check that we can actually afford the new fee compare to the holder balance.
+		if let Some(reserve) = self.counterparty_selected_channel_reserve_satoshis {
+			let num_htlcs = self.pending_inbound_htlcs.len() + self.pending_outbound_htlcs.len();
```

Comment:
I believe we should include the holding cell for outbound. Also, maybe better to just use `get_inbound/outbound_pending_htlc_stats` here?